### ISSUE TITLE
Product option values not always integers (fails to load labels)

### ISSUE THREAD

**Issue by simonmaass** [DivanteLtd/vue-storefront#4851]
## Current behavior

in core\modules\catalog\helpers\productOptions\getSelectedOption.ts  line 13

`const selectedOptionMeta = option.values.find(ov => ov.value_index === selectedOption.value)`

This will only work if value and value_index are both integers or strings... 

Currently using vsbridge and getting configurableOptions the values as strings and selectedOption as int:

```
option {
  attribute_id: 178,
  attribute_code: 'size',
  label: 'Größe',
  values: [
    { value_index: '273', label: 'S' },
    { value_index: '274', label: 'M' },
    { value_index: '275', label: 'L' }
  ]
} 
```
## Expected behavior

We should probably convert both to string or int....

## Version of Vue Storefront

- [x] Vue Storefront
- [ ] Vue Storefront Next

## Can you handle fixing this bug by yourself?

- [x] YES
- [ ] NO

## Which [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html) state this refers to? Info for developer. (doesn't apply to Next)
Pick one option.

- [ ] This is a bug report for test version on https://test.storefrontcloud.io - In this case Developer should create branch from `develop` branch and create Pull Request `2. Feature / Improvement` back to `develop`.
- [ ] This is a bug report for current Release Candidate version on https://next.storefrontcloud.io - In this case Developer should create branch from `release` branch and create Pull Request `3. Stabilisation fix` back to `release`.
- [ ] This is a bug report for current Stable version on https://demo.storefrontcloud.io and should be placed in next stable version hotfix - In this case Developer should create branch from `hotfix` or `master` branch and create Pull Request `4. Hotfix` back to `hotfix`.

## Environment details

- Browser: 
- OS: 
- Node: 
- Code Version:  1.12.3

## Additional information

This should do the job...

`const selectedOptionMeta = option.values.find(ov => ov.value_index.toString() === selectedOption.value.toString())`

**Comment by simonmaass** [comment]
PR created

**Comment by Fifciu** [comment]
Solved in #4875
